pilot/pkg/proxy/envoy/v1/discovery.go

```diff
@@ -481,6 +489,19 @@ func (ds *DiscoveryService) ClearCacheStats(_ *restful.Request, _ *restful.Respo
 // clearCache will clear all envoy caches. Called by service, instance and config handlers.
 // This will impact the performance, since envoy will need to recalculate.
 func (ds *DiscoveryService) clearCache() {
+	clearCacheMutex.Lock()
+	defer clearCacheMutex.Unlock()
+	if time.Since(lastClearCache) < 60*time.Second {
+		if !clearCacheTimerSet {
+			clearCacheTimerSet = true
+			time.AfterFunc(61*time.Second, func() {
```

Comment:
isn't this going to block the event queue?

Reply:
the danger here is the unbound growth of the incoming events. i think we need to run a secondary worker thread to refresh the cache like @rshriram suggested. they only need to share one bit for the "dirtiness" of the cache.